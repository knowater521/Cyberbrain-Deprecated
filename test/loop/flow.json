[
  {
    "code": "for i in range(3):",
    "next": "print(i)",
    "prev": "",
    "step_into": "",
    "returned_from": ""
  },
  {
    "code": "print(i)",
    "next": "for i in range(3):",
    "prev": "for i in range(3):",
    "step_into": "",
    "returned_from": ""
  },
  {
    "code": "for i in range(3):",
    "next": "print(i)",
    "prev": "print(i)",
    "step_into": "",
    "returned_from": ""
  },
  {
    "code": "print(i)",
    "next": "for i in range(3):",
    "prev": "for i in range(3):",
    "step_into": "",
    "returned_from": ""
  },
  {
    "code": "for i in range(3):",
    "next": "print(i)",
    "prev": "print(i)",
    "step_into": "",
    "returned_from": ""
  },
  {
    "code": "print(i)",
    "next": "for i in range(3):",
    "prev": "for i in range(3):",
    "step_into": "",
    "returned_from": ""
  },
  {
    "code": "for i in range(3):",
    "next": "print(\"in else\")",
    "prev": "print(i)",
    "step_into": "",
    "returned_from": ""
  },
  {
    "code": "print(\"in else\")",
    "next": "while i > 0:",
    "prev": "for i in range(3):",
    "step_into": "",
    "returned_from": ""
  },
  {
    "code": "while i > 0:",
    "next": "i -= 1",
    "prev": "print(\"in else\")",
    "step_into": "",
    "returned_from": ""
  },
  {
    "code": "i -= 1",
    "next": "if i == 1:",
    "prev": "while i > 0:",
    "step_into": "",
    "returned_from": ""
  },
  {
    "code": "if i == 1:",
    "next": "break",
    "prev": "i -= 1",
    "step_into": "",
    "returned_from": ""
  },
  {
    "code": "break",
    "next": "cyberbrain.register(i)",
    "prev": "if i == 1:",
    "step_into": "",
    "returned_from": ""
  },
  {
    "code": "cyberbrain.register(i)",
    "next": "",
    "prev": "break",
    "step_into": "",
    "returned_from": ""
  }
]