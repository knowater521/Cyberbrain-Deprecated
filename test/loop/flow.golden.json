[
  {
    "code": "for i in range(3):",
    "next": "print(i)",
    "prev": "",
    "step_into": "",
    "returned_from": "",
    "var_changes": [
      "appear i=0\n"
    ],
    "location": "loop.py: <module>"
  },
  {
    "code": "print(i)",
    "next": "for i in range(3):",
    "prev": "for i in range(3):",
    "step_into": "",
    "returned_from": "",
    "location": "loop.py: <module>",
    "tracking": [
      "i"
    ]
  },
  {
    "code": "for i in range(3):",
    "next": "print(i)",
    "prev": "print(i)",
    "step_into": "",
    "returned_from": "",
    "var_changes": [
      "modify i 0 -> 1\n"
    ],
    "location": "loop.py: <module>",
    "tracking": [
      "i"
    ]
  },
  {
    "code": "print(i)",
    "next": "for i in range(3):",
    "prev": "for i in range(3):",
    "step_into": "",
    "returned_from": "",
    "location": "loop.py: <module>",
    "tracking": [
      "i"
    ]
  },
  {
    "code": "for i in range(3):",
    "next": "print(i)",
    "prev": "print(i)",
    "step_into": "",
    "returned_from": "",
    "var_changes": [
      "modify i 1 -> 2\n"
    ],
    "location": "loop.py: <module>",
    "tracking": [
      "i"
    ]
  },
  {
    "code": "print(i)",
    "next": "for i in range(3):",
    "prev": "for i in range(3):",
    "step_into": "",
    "returned_from": "",
    "location": "loop.py: <module>",
    "tracking": [
      "i"
    ]
  },
  {
    "code": "for i in range(3):",
    "next": "print(\"in else\")",
    "prev": "print(i)",
    "step_into": "",
    "returned_from": "",
    "location": "loop.py: <module>",
    "tracking": [
      "i"
    ]
  },
  {
    "code": "print(\"in else\")",
    "next": "while i > 0:",
    "prev": "for i in range(3):",
    "step_into": "",
    "returned_from": "",
    "location": "loop.py: <module>",
    "tracking": [
      "i"
    ]
  },
  {
    "code": "while i > 0:",
    "next": "i -= 1",
    "prev": "print(\"in else\")",
    "step_into": "",
    "returned_from": "",
    "location": "loop.py: <module>",
    "tracking": [
      "i"
    ]
  },
  {
    "code": "i -= 1",
    "next": "if i == 1:",
    "prev": "while i > 0:",
    "step_into": "",
    "returned_from": "",
    "var_changes": [
      "modify i 2 -> 1\n"
    ],
    "location": "loop.py: <module>",
    "tracking": [
      "i"
    ]
  },
  {
    "code": "if i == 1:",
    "next": "break",
    "prev": "i -= 1",
    "step_into": "",
    "returned_from": "",
    "location": "loop.py: <module>",
    "tracking": [
      "i"
    ]
  },
  {
    "code": "break",
    "next": "cyberbrain.register(i)",
    "prev": "if i == 1:",
    "step_into": "",
    "returned_from": "",
    "location": "loop.py: <module>",
    "tracking": [
      "i"
    ]
  },
  {
    "code": "cyberbrain.register(i)",
    "next": "",
    "prev": "break",
    "step_into": "",
    "returned_from": "",
    "location": "loop.py: <module>",
    "tracking": [
      "i"
    ]
  }
]